keras_cv/layers/preprocessing/erasing.py

```diff
@@ -0,0 +1,196 @@
+    def _compute_patch_size(self, batch_size, image_height, image_width):
+        pass
+
+
+class RandomErasing(BaseErasing):
+    """RandomErasing implements the RandomErasing data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        scale: Tuple of float. Area scale range (min, max) of erasing patch.
+        ratio: Tuple of float. Aspect ratio range (min, max) of erasing patch.
+        patch_value: Float or string. The value to fill in the patches. If "gaussian", will
+            fill patches with gaussian noise. Defaults to "gaussian".
+    References:
+       [RandomErasing paper](https://arxiv.org/abs/1708.04896).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    random_erase = keras_cv.layers.preprocessing.erasing.RandomErasing(1.0)
+    augmented_images, labels = random_erase(images, labels)
+    ```
+    """
+
+    def __init__(
+        self,
+        rate,
+        scale=(0.02, 0.33),
+        ratio=(0.3, 3.3),
+        patch_value="gaussian",
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(rate, patch_value, seed, **kwargs)
+        self.scale = scale
+        self.ratio = ratio
+
+    def _compute_patch_size(self, batch_size, image_height, image_width):
+        area = tf.cast(image_height * image_width, tf.float32)
+        erase_area = area * tf.random.uniform(
+            [batch_size], minval=self.scale[0], maxval=self.scale[1]
+        )
+        aspect_ratio = tf.random.uniform(
+            [batch_size], minval=self.ratio[0], maxval=self.ratio[1]
+        )
+        h = tf.cast(tf.round(tf.sqrt(erase_area * aspect_ratio)), tf.int32)
+        w = tf.cast(tf.round(tf.sqrt(erase_area / aspect_ratio)), tf.int32)
+
+        h = tf.minimum(h, image_height - 1)
+        w = tf.minimum(w, image_width - 1)
+
+        return h, w
+
+    def get_config(self):
+        config = {
+            "scale": self.scale,
+            "ratio": self.ratio,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+class CutOut(BaseErasing):
+    """CutOut implements the CutOut data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        length: Integer. The side length of the square patches to cut out.
+        patch_value: Float or string. The value to fill in the patches. If "gaussian", will
+            fill patches with gaussian noise. Defaults to 0.0.
+    References:
+       [CutOut paper](https://arxiv.org/abs/1708.04552).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    cutout = keras_cv.layers.preprocessing.erasing.CutOut(1.0, 50)
+    augmented_images, labels = cutout(images, labels)
+    ```
+    """
+
+    def __init__(self, rate, length, patch_value=0.0, seed=None, **kwargs):
```

Comment:
Maybe we should actually make length a float from 0-1 that represents the relative width/height of the image?

So for an image of (150, 150) a length of 0.5 is equivalent to 75.

Reply:
(this applies to both classes)

Also open to discussion here, @fchollet may have some thoughts.  One benefit of using a float in range [0, 1] is that it can be copy pasted into new code bases in a meaningful way.  As a reader the number 0.4 is also very meaningful to me: 40% of the image.  40 could mean anything from covering the entire image to covering 0.01% of a high res image.

Reply:
maybe `size`, or `width` would be good as well?

Reply:
I can follow your logic. A scale between [0, 1] would also make `CutOut` follow the same pattern as `RandomErasing`, increasing consistency.

However, the [CutOut paper](https://arxiv.org/pdf/1708.04552.pdf) only denotes sizes in terms of absolute values, while the [RandomErasing paper](https://arxiv.org/pdf/1708.04896.pdf) actually denotes sizes in relative values. If a user wants to draw references to the papers, it might be confusing for users if the size formats differ.

Perhaps we should weigh consistency higher?

Reply:
I'd prefer to weight consistency higher here.  We may also want to use max_aspect and min_aspect.

I do quite like the name aspect.

Reply:
Actually, CutOut is always a square patch per the official paper. It will be confusing to specify a square patch using a scalar between [0, 1] as relative width/height of input image, if input image is non-square.

Reply:
Ah very interesting.  I don't personally see a reason we need to match the implementation in that regard - though perhaps @fchollet may?

Reply:
> Ah very interesting. I don't personally see a reason we need to match the implementation in that regard - though perhaps fchollet may?

@LukeWood Let me know if I should make any changes to how the patch size is specified, and what change you would prefer. Unless we are waiting for fchollet's opinion?

Reply:
Yeah, let's wait for @fchollet 's opinion here.  He also mentioned that we may want to merge the layers and rely just on the fill arg.  He said he would do a pass on this PR in a bit.